packages/firefuel/lib/src/firefuel_collection.dart

```diff
@@ -60,17 +60,14 @@ abstract class FirefuelCollection<T extends Serializable>
   );
 
   @override
-  Stream<T?> listen<T>(
-    CollectionReference<T?> ref,
-    DocumentId docId,
-  ) {
+  Stream<T?> listen(DocumentId docId) {
     return ref.doc(docId.docId).snapshots().map(
           (snapshot) => snapshot.data(),
         );
   }
 
   @override
-  Stream<List<T>> listenAll<T>(CollectionReference<T?> ref) {
+  Stream<List<T>> listenAll(CollectionReference<T?> ref) {
```

Comment:
We need to accept ref as an argument. We already have the property in the class.

```suggestion
  Stream<List<T>> listenAll() {
```